liminal/core/config/config.py

```diff
@@ -0,0 +1,146 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+import traceback
+
+from liminal.core.config.defaults import hyperliminal, default_configs
+from liminal.core.util import dict_util
+from liminal.core.util import files_util
+
+
+class ConfigUtil:
+    """
+    Load and enrich config files under configs_path.
+    """
+    __HYPERLIMINAL = "hyperliminal"
+    __PIPELINES = "pipelines"
+    __SUPER = "super"
+    __TYPE = "type"
+    __SUB = "sub"
+    __SERVICES = "services"
+    __TASKS = "tasks"
+    __PIPELINE_DEFAULTS = "pipeline_defaults"
+
+    def __init__(self, configs_path):
+        self.configs_path = configs_path
+        self.config_files = files_util.load(configs_path)
+        self.hyperliminal = hyperliminal.HYPERLIMINAL
+        self.loaded_subliminals = []
+
+    def safe_load(self, is_render_variables):
+        """
+        :returns list of config files after enrich with defaults and supers
+        """
+        if self.loaded_subliminals:
+            return self.loaded_subliminals
+
+        configs = self.config_files.values()
+        enriched_configs = []
+
+        for subliminal in [config for config in configs if self.__is_subliminal(config)]:
+            name = subliminal.get('name')
+            logging.info(f'Loading yml {name}')
+            # noinspection PyBroadException
+            try:
+                superliminal = self.__get_superliminal(subliminal)
+                enriched_config = self.__merge_configs(subliminal, superliminal,
+                                                       is_render_variables)
+                enriched_configs.append(enriched_config)
+            except Exception:
+                logging.error(f'Failed to load yml {name}')
+                traceback.print_exc()
+
+        self.loaded_subliminals = enriched_configs
```

Comment:
is there a way to  see the result of the "merged" liminal files as output?
similar to kustomize build ? useful to test/verify that the compilation was correct?
Perhaps it makes sense also to deploy the merged yamls from file system (i.g. they are part of the deploy cli)?

Reply:
You right. We are working on it in our fork. I will add it here once we will finish.

Reply:
@assapin done